config/kube.py

```diff
@@ -207,33 +298,63 @@ def kube_reset():
                 ignore_error=True)
 
     clicommon.run_command("kubeadm reset -f", ignore_error=True)
-    clicommon.run_command("rm -rf /etc/cni/net.d")
-    clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    clicommon.run_command("rm -rf {}".format(CNI_NET_DIR))
+    if purge_conf:
+        clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
     clicommon.run_command("systemctl stop kubelet")
     clicommon.run_command("systemctl disable kubelet")
 
+    _update_kube_server_state(state_db, False)
+
+
+def kube_reset(force=False):
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    state_db = _get_state_db()
+    if not force:
+        if not is_connected():
+            # Already *not* connected. No-Op. Just ensure SERVER-DB is updated
+            _update_kube_server_state(state_db, False)
+            return
+
+    _do_reset(state_db, True)
+
 
 def kube_join(force=False):
     lock_fd = _take_lock()
     if not lock_fd:
         log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
         return
 
-    db_data = Db().get_data('KUBERNETES_MASTER', 'SERVER')
-    if not db_data or 'IP' not in db_data or not db_data['IP']:
+    db_data = {
+            KUBE_SERVER_IP: "",
+            KUBE_SERVER_DISABLE: "false"
+            }
+    db_data.update(Db().get_data(KUBE_SERVER_TABLE_NAME, KUBE_SERVER_TABLE_KEY))
```

Comment:
When try to join unconfigured master server, I hit this error: 
```  
File "/usr/local/lib/python2.7/dist-packages/config/kube.py", line 336, in kube_join
    db_data.update(Db().get_data(KUBE_SERVER_TABLE_NAME, KUBE_SERVER_TABLE_KEY))
TypeError: 'NoneType' object is not iterable
```

Reply:
No more kube.join